### PR TITLE
Setup marathon speedup

### PR DESCRIPTION
**Pull request by mattmb**
Just scratching an itch with this PR. I've attempted to pin down some slow bits in `setup_marathon_job` and make them "better".

Not really sure how much time this might save. The getting of marathon tasks will only improve if we call this script with multiple services (which we currently don't). @mjksmith you're the expert on this right? Do you think this will help?

### CODE REVIEW

**Comment by mattmb** [comment]
Ug, looks like the itests have failed, I'll take a look and see if I've broken anything "real"

File: paasta_tools/setup_marathon_job.py [hunk 1]
```diff
@@ -241,7 +241,8 @@ def log_bounce_action(line, level='debug'):
         at_risk_tasks=bounce_lib.flatten_tasks(old_app_at_risk_tasks),
     )
 
-    kill_given_tasks(client=client, task_ids=[task.id for task in tasks_to_kill], scale=True)
+    if tasks_to_kill:
+        kill_given_tasks(client=client, task_ids=[task.id for task in tasks_to_kill], scale=True)
```
Comment (mjksmith) [inline paasta_tools/setup_marathon_job.py]: It seems better that `kill_given_tasks` knows how to do 'the right thing' and doesn't call `client.kill_given_tasks` if `tasks_to_kill` is empty.

**Comment by solarkennedy** [comment]
Ship!

File: paasta_tools/setup_marathon_job.py (hunk 1 above)
Comment (mattmb) [inline paasta_tools/setup_marathon_job.py]: That's a good shout, I'll move this fix into the function itself.

**Comment by mattmb** [comment]
@mjksmith I've only profiled it on my local docker-compose setup. It certainly makes some improvement but it's hard to measure in such a fake environment!

**Pull request merged**